### ISSUE TITLE
Jigsaw Puzzle restore

### ISSUE THREAD

**Issue by doem97** [yingyichen-cyy/JigsawViT#3]
Dear authors,

Is your jigsaw restore performance well? For ImageNet/jigsaw_deit files, I directly use your pre-trained model to restore my permutated images, but the performance seems not good. Can you show us some restored images from permutation? Many thanks.
![image](https://github.com/yingyichen-cyy/JigsawViT/assets/19631039/d550bfed-c65a-4b15-8ec6-8acb1ac2a5ad)
![image](https://github.com/yingyichen-cyy/JigsawViT/assets/19631039/e703ba81-95e1-43fb-8f9a-0c6542fcbfe3)
![image](https://github.com/yingyichen-cyy/JigsawViT/assets/19631039/910eb350-e53b-42b8-b69e-3e73e33ed34b)

**Comment by yingyichen-cyy** [comment]
Dear doem97,
I find that there may be some misunderstandings towards our jigsaw methods in the paper. We do not need to manually implement random permutation for the input image. We use "patch masking" in the paper and there are two main modifications: (1) we randomly mask out some of the input patches, (2) for the jigsaw branch, we omit the positional embedding, hence providing no hints of the position to the jigsaw branch. Although the second one is not really random permutation of image patches, it can in some degree serve the similar purpose. 

Hence, if you send a permuted image to Jigsaw-ViT, the ground-truth jigsaw targets will still be [0,1,2,...,195] as shown in the following image. To be more exact, _the first patch in the permuted image will be the new first patch with jigsaw ground-truth target 0._ That is why your restored image looks similar to the permuted image since the jigsaw prediction should restore the absolute position of the permuted image. Hope these can resolve your concerns.
![p](https://github.com/yingyichen-cyy/JigsawViT/assets/48011560/f022114e-1b67-44ca-837e-717fcbe4ec55)
